Documentation for changes related to 'You are already logged in' scen… (#28595)
keycloak/keycloak: docs/documentation/release_notes/topics/25_0_0.adoc

memory, which is a requirement to be resistant against GPU attacks. The defaults for Argon2 in {project_name} requires 7MB
per-hashing request.

= Cookies updates

== SameSite attribute set for all cookies

The following cookies did not use to set the `SameSite` attribute, which in recent browser versions results in them
defaulting to `SameSite=Lax`:

* `KC_STATE_CHECKER` now sets `SameSite=Strict`
* `KC_RESTART` now sets `SameSite=None`
* `KEYCLOAK_LOCALE` now sets `SameSite=None`
* `KEYCLOAK_REMEMBER_ME` now sets `SameSite=None`

The default value `SameSite=Lax` causes issues with POST based bindings, mostly applicable to SAML, but also used in
some OpenID Connect / OAuth 2.0 flows.

== Removing KC_AUTH_STATE cookie

> The cookie `KC_AUTH_STATE` was removed and it is not set anymore by {project_name} server as server does not need this cookie anymore.

Reviewer: ```suggestion
The cookie `KC_AUTH_STATE` is removed and it is no longer set by the {project_name} server as this server no longer needs this cookie.
```

* `HttpCookie` - replaced by `NewCookie.Builder`
* `HttpResponse.setCookieIfAbsent(HttpCookie cookie)` - replaced by `HttpResponse.setCookieIfAbsent(NewCookie cookie)`

= Addressed 'You are already logged in' for expired authentication sessions

> In Keycloak 23 release, there was improvement added for the case when user authenticated in parallel in multiple browser tabs. However it did not handled well the case when authentication session
> expired. Now for the case when user is already logged-in in one browser tab and authentication session expired in other browser tabs, {project_name} is able to redirect back to the client

Reviewer: ```suggestion
The Keycloak 23 release provided improvements for when a user is authenticated in parallel in multiple browser tabs. However, this improvement did not address the case when authentication session
expired. Now for the case when user is already logged-in in one browser tab and authentication session expired in other browser tabs, {project_name} is able to redirect back to the client
```

> application with some OIDC/SAML error, so client application can immediately retry authentication, which should usually automatically login the application because of the SSO session. More

Reviewer: ```suggestion
application with a OIDC/SAML error, so the client application can immediately retry authentication, which should usually automatically log in the application because of the SSO session. More
```

> details in the link:{adminguide_link}#_authentication-sessions[{adminguide_name} authentication sessions chapter].

Reviewer: ```suggestion
details, see link:{adminguide_link}#_authentication-sessions[{adminguide_name} authentication sessions].
```
